keyboards/thepanduuh/degenpad/degenpad.c

```diff
@@ -0,0 +1,17 @@
+/*
+Copyright 2023 ThePanduuh <thepanduuh.kb@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "degenpad.h"
```

Comment:
This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "degenpad.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

Reply:
Would it make sense to leave the ENCODER_MAP_ENABLE setting in the rules.mk file for the default keymap, and leaving the Encoder layer definition in the keymap with this change?

Reply:
And also, if this is in the keyboard.c file, if I map the encoder to something else in VIA, will it cause issues? Or will the VIA map take priority over this?

Reply:
> Would it make sense to leave the ENCODER_MAP_ENABLE setting in the rules.mk file for the default keymap, and leaving the Encoder layer definition in the keymap with this change?

Yes, that's just fine.

> And also, if this is in the keyboard.c file, if I map the encoder to something else in VIA, will it cause issues? Or will the VIA map take priority over this?

The latter, but if you want to make it extra intuitive, you could do 
```c
#if defined(ENCODER_ENABLE) && !defined(ENCODER_MAP_ENABLE)
//...
#endif
```

Reply:
Okay, that's good to know. Right now, I have a rules.mk file in the default keymap folder to enable encoder mapping. Should I remove this, because to me it seems like it adds little/no value as the definition is now done in this file vs previously where I used the keymap file to define it and there's no mapping of encoder in the keymap.c file anymore.
Trying to think of the best practice/least amount of excess code.